packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -32,7 +32,7 @@ export 'message_codec.dart' show MessageCodec, MethodCall, MethodCodec;
 /// The statistics include the total bytes transmitted and the average number of
 /// bytes per invocation in the last quantum. "Up" means in the direction of
 /// Flutter to the host platform, "down" is the host platform to flutter.
-const bool kProfilePlatformChannels = false;
+bool debugProfilePlatformChannels = false;
```

Comment:
So what @jiahaog was referring to is that previously this would get tree shaken.  My concern is adding overhead for release builds.  In a previous PR Jia Hao made it possible to turn this on for release builds when it used to just be for profile builds or debug builds.  We also have the convention that `debug*` variables and procedures should be tree-shaken in release builds but it can't happen the way this is written.

I think what we can do to solve that is:
1) Keep `debugProfilePlatformChannels` but only access it in `!kRelease` branches.
2) Keep `const bool kProfilePlatformChannels = false;`
3) Make the decision to use them be like this

```dart
if (kProfilePlatformChannels || (!kRelease && debugProfilePlatformChannels) {}
```

That way it can be turned on and off in profile/debug at runtime and can be turned on for release builds at compile time.  What do you think @jiahaog @kenzieschmoll ?

Reply:
I implemented these suggestions and added a convenience getter. PTAL.